### DISCUSSION TITLE
Only show tree view if extension is activated

### DISCUSSION THREAD

**Discussion by itavero** [microsoft/vscode-discussions#594] [Extension Development QnA] [answered]
My extension contributes a tree view in the Explorer tab. My extension is only activated if certain files are found in the workspace (it actually depends on another extension being activated, but I haven't found a way yet to express that in `packages.json`, so I'll open up another discussion for that).

Currently the view is shown in the Explorer tab, even if the extension is not activated (e.g. even if I don't have a workspace open at all or if the files are not present).
How can I prevent this?

I'm assuming I should do something with the `where` clause. Unfortunately, I think the [tree view sample](https://github.com/Microsoft/vscode-extension-samples/blob/main/tree-view-sample/USAGE.md) isn't very clear on how this should actually be used.
It mentions an example with `where` set to `workspaceHasPackageJSON`, but I can't find `workspaceHasPackageJSON` anywhere else in that repository (nor in the vscode main repo).
(PS: I'm using the "new" search view from GitHub, so it could potentially be that it has a bug that is preventing me from finding this, although that's perhaps unlikely).

Hopefully someone has a reference for me on how to achieve this.

**Comment by r3m0t** [comment]
You can use the `setContext` command and a FileSystemWatcher + onDidChangeWorkspaceFolders to update the context when files are created or deleted. https://code.visualstudio.com/api/references/when-clause-contexts

**Reply by itavero** [reply, marked as answer]
Thanks for the pointer. Based on this I did the following to make sure that the view is only shown when my extension is active.

In the `activate` method of my extensions, I now do something like:
```ts
  vscode.commands.executeCommand(
    "setContext",
    "my-ext.showTreeView",
    true
  );
```

And in my extensions `deactivate` method, I do the same but then set it to `false` instead.
In the `when` clause of the view, I simply put `my-ext.showTreeview`.
